bin/tx-prover/README.md

```diff
@@ -120,6 +122,28 @@ Then access the Jaeger UI at `http://localhost:16686/`.
 
 If Docker is not an option, Jaeger can also be set up directly on your machine or hosted in the cloud. See the [Jaeger documentation](https://www.jaegertracing.io/docs/) for alternative installation methods.
 
+## Metrics
+
+The proxy includes a service that provides metrics for [Prometheus](https://prometheus.io/docs/introduction/overview/). This service is always enabled and uses the port defined in the configuration file. The metrics are available at the `/metrics` endpoint.
+
+To consume and display the metrics, you can use Prometheus and Grafana. The simplest way to install Prometheus and Grafana is by using Docker containers. To do so, run:
```

Comment:
Let's explain in more detail how this actually works (i.e., Prometheus collects the metrics and Grafana displays them).

Btw, one thing that is not clear to me: do the metrics get re-set after `/metrics` endpoint is called? I'm assuming not, but then how do we differentiate between metrics that are point-in-time measurements (e.g., number of workers) and metrics that are interval measurements (e.g., number of requests).

Reply:
> do the metrics get re-set after /metrics endpoint is called? I'm assuming not but then how do we differentiate between metrics that are point-in-time measurements (e.g., number of workers) and metrics that are interval measurements (e.g., number of requests).

You are right here, the metrics state is kept throughout the service life, unless we explicitly reset a metrics in the code (e.g. calling `WORKER_COUNT.reset()` which sets the value back to zero).

Since Prometheus when scraps the metrics adds a timestamp, when then can use rates to measure how a metric vary over time. 

Using number of requests as example, every time a new request is received we increase an accumulator (can only go up). If we read this metric directly when can have the number of total request that we received since the beginning of the server, but if we create a query benefiting from the prometheus timeseries DB using something like `rate(request_count[1m])` , we can tell how many requests per second were handled, on average, during that time window.

I'm expanding the docs with some more info about the whole process.